docs/DeveloperGuide.md

```diff
@@ -0,0 +1,128 @@
+- [Frontend Components](#frontend-components)
+- [Dashboard Generator Components](#dashboard-generator-components)
+- [Git Utilities](#git-utilities)
+- [Analyzer Components](#analyzer-components)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+
+1. **JDK `1.8.0_60`**  or later
+2. **Git** on the command line 
+3. **findstr** for Windows, **grep** for macOS or Linux on the command line
+   * Ensure that you're able to use these tools on the OS terminal.
+4. **IntelliJ** IDE [Optional]
+
+### Setting up the project in your computer using IntelliJ
+1. Fork this repo, and clone the fork to your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle.
+4. Click `Configure` > `Project Defaults` > `Project Structure`.
+5. Click `New…​` and find the directory of the JDK.
+6. Click `Import Project`.
+7. Locate the build.gradle file and select it. Click OK.
+8. Ensure that the selected version of `Gradle JVM` matches our prerequisite.
+9. Click `OK` to accept the all the other default settings.
+
+#### Verifying the setup
+1. Ensure that Gradle build without error.
+2. Run the tests to ensure they all pass.
+   1. Go to `src` -> `main` -> `test`. Right click -> Run Tests.
+
+#### Configuring the coding style
+This project follows [oss-generic coding standards](https://oss-generic.github.io/process/docs/CodingStandards.html). IntelliJ’s default style is mostly compliant with our Java coding convention but it uses a different import order from ours. To rectify,
+
+1. Go to `File` > `Settings…`​ (Windows/Linux), or `IntelliJ IDEA` > `Preferences…`​ (macOS)
+2. Select `Editor` > `Code Style` > `Java`
+3. Click on the `Imports` tab to set the order
+   * For `Class count to use import with '*'` and `Names count to use static import with '*'`: Set to `999` to prevent IntelliJ from contracting the import statements
+   * For `Import Layout`: The order is `import static all other imports`, `import java.*`, `import javax.*`, `import org.*`, `import com.*`, `import all other imports`. Add a ``<blank line>`` between each `import`
+
+Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document to configure Intellij to check style-compliance as you write code.
+
+### Before writing code
+1. Do check out our [process guide](../docs/Process.md) before start making changes.
+
+## Architecture
+
+ ![architecture](images/architecture.png)
+
+ Above is the proposed overall architecture of RepoSense.<br>
+ Below gives a brief description for each of `RepoSense`'s component. 
+
+
+## Model
+Model holds the data structures that are commonly used by the different aspects of RepoSense.
+
+### Author
+`Author` stores the single GitHub ID of the author. Any contribution or commits of the author which may have different aliases will be attributed to the same `Author` object.
```

Comment:
Perhaps `stores the GitHub ID and the aliases of an author`?

Any contribution or commits ~of the author which may have different aliases will be attributed to the same `Author` object.~ `made by the author, using his/her GitHub ID or aliases, will be attributed to the same Author object.`